### ISSUE TITLE
Generics.Deriving.Uniplate documentation

### ISSUE THREAD

**Issue by mgajda** [dreixel/generic-deriving#88]
It would be nice to have a documentation instead of constantly referring to old `uniplate` package.

Would you accept a patch copy-pasting it?

**Comment by RyanGlScott** [comment]
Yes, I'd happily accept a patch to expand upon the documentation.
